lua/muren/ui.lua

```diff
@@ -249,17 +249,65 @@ local get_nvim_ui_size = function()
   end
 end
 
+local make_ui_positions = function()
+  -- NOTE assumes that options will be populated by options.populate() before this is called,
+  -- as the open_preview function does
+  local vertical_anchor = options.default.vertical_anchor
+  local horizontal_anchor = options.default.horizontal_anchor
+  local vertical_offset = options.default.vertical_offset
+  local horizontal_offset = options.default.horizontal_offset
+  local gheight, gwidth = get_nvim_ui_size()
+  local positions = {
+    row = {
+      top = 0,
+      center = (gheight - options.values.total_height) / 2,
+      bottom = (gheight - options.values.total_height),
+    },
+    col = {
+      left = 0,
+      center = (gwidth - options.values.total_width) / 2,
+      right = (gwidth - options.values.total_width),
+    },
+  }
+  local adjusted_row = positions.row[vertical_anchor] + vertical_offset
+  local adjusted_col = positions.col[horizontal_anchor] + horizontal_offset
```

Comment:
(writing this with an eye to the nui discussion in #24 ) do you mean something like, change the calculation so the user setting for an offset is always a positive value? e.g. if h_anchor is 'right', then subtract the (positive) value of the h_offset, else if h_anchor is 'left', add it? (and similarly for v_anchor). This way the offset is always referencing how far off it is relative the anchor instead of being a shift relative to the screen as a whole.

Or did you mean something else?

Reply:
Yes exactly :)